share/getters/ipld.go

```diff
@@ -76,6 +77,10 @@ func (ig *IPLDGetter) GetEDS(ctx context.Context, root *share.Root) (eds *rsmt2d
 		// convert error to satisfy getter interface contract
 		err = share.ErrNotFound
 	}
+	var errByz *byzantine.ErrByzantine
+	if errors.As(err, &errByz) {
+		return nil, err
+	}
```

Comment:
Wait yes, why? Does this not already happen inside the retriever session?

Reply:
I don't understand why we would short circuit here

Reply:
please take a look at the next few lines:
`if err != nil -> return nil, fmt.Errorf("getter/ipld: failed to retrieve eds: %w", err)`. So I won't identify byzErr then

Reply:
I thought errors.Is identifies wrapped errors like this 🤔

https://go.dev/blog/go1.13-errors

Maybe its something in the stack above? Are we using errors.Is?

Reply:
You cannot parse wrapped errors with error.As and in case of byzErr we need the underlying data(as this is not a simple error but a struct  that implements error interface)

Reply:
Ah okay 👍🏼 makes sense

Reply:
Also,
```
var errByz *byzantine.ErrByzantine
if errors.Is(err, errByz)
```

won't work because errByz is a struct.